src/adapters/lokijs/worker/lokiWorker.js

```diff
@@ -51,29 +49,55 @@ export default class LokiWorker {
 
   executeNext(): void {
     const action = this.queue[0]
-    const onActionDone = (response: WorkerResponse): void => {
-      invariant(this._actionsExecuting === 1, 'worker queue should have 1 item')
-      this._actionsExecuting = 0
-      this.queue.shift()
+    invariant(this._actionsExecuting === 0, 'worker should not have ongoing actions') // sanity check
+    this.processAction(action)
+  }
 
-      this.workerContext.postMessage(response)
+  onActionDone(response: WorkerResponse): void {
+    invariant(this._actionsExecuting === 1, 'worker should be executing 1 action') // sanity check
+    this._actionsExecuting = 0
+    this.queue.shift()
 
-      if (this.queue.length) {
-        this.executeNext()
-      }
+    try {
+      this.workerContext.postMessage(response)
+    } catch (error) {
+      logError(error)
     }
 
-    invariant(this._actionsExecuting === 0, 'worker queue should be empty') // sanity check
-    this.processAction(action, onActionDone)
+    if (this.queue.length) {
+      this.executeNext()
+    }
   }
 
-  async processAction(action: WorkerAction, callback: WorkerResponse => void): Promise<void> {
+  processAction(action: WorkerAction): void {
     try {
       this._actionsExecuting += 1
 
       const { type, payload, id } = action
       invariant(type in actions, `Unknown worker action ${type}`)
 
+      if (type === actions.SETUP || type === actions.UNSAFE_RESET_DATABASE) {
+        this.processActionAsync(action)
+      } else {
+        // run action
+        invariant(this.executor, `Cannot run actions because executor is not set up`)
+
+        const runExecutorAction = executorMethods[type].bind(this.executor)
+        const response = runExecutorAction(...payload)
```

Comment:
Hmm it's very similar to else block in `processActionAsync`. The only difference is `await` keyword. Maybe move this to helper method to avoid code duplication?

Reply:
✅